### PR TITLE
Improve incentive sync documentation

### PR DESCRIPTION
**Pull request by rhuairahrighairidh**
- Adds a desciption of the math behind the sync methods in a more abstract way than is represented in code.
- Enumerates input possibilities to try and highlight edge cases. For example when new reward period is added for pre-existing deposits.

### CODE REVIEW

**Review by galxy25 — APPROVED** [review]

File: x/incentive/spec/01_concepts.md [hunk 1]
```diff
@@ -4,23 +4,86 @@ order: 1
 
 # Concepts
 
-This module presents an implementation of user incentives that are controlled by governance. When users take a certain action, for example opening a CDP, they become eligible for rewards. Rewards are **opt in** meaning that users must submit a message before the claim deadline to claim their rewards. The goals and background of this module were subject of a previous Kava governance proposal, which can be found [here](https://ipfs.io/ipfs/QmSYedssC3nyQacDJmNcREtgmTPyaMx2JX7RNkMdAVkdkr/user-growth-fund-proposal.pdf)
+This module implements governance controlled user incentives. When users take a certain action, for example opening a CDP, they become eligible for rewards. Rewards are **opt in** meaning that users must submit a message before the claim deadline to claim their rewards. The goals and background of this module were subject of a previous Kava governance proposal, which can be found [here](https://ipfs.io/ipfs/QmSYedssC3nyQacDJmNcREtgmTPyaMx2JX7RNkMdAVkdkr/user-growth-fund-proposal.pdf)
```
[the remark this answers is not in the document]
Comment (galxy25) [reply x/incentive/spec/01_concepts.md:7]: (thinking performance wise how variable number of claiming actions can impact block time and whether it's possible to safely process claims concurrently across claimers or async between blocks)

File: x/incentive/spec/01_concepts.md [hunk 2]
```diff
@@ -4,23 +4,86 @@ order: 1
...
 
-Every block a global tracker adds up total rewards paid out per unit of user involvement. For example, per unit of xrpb supplied to hard, or per share in a kava/usdx swap pool. A user's specific reward can then be calculated as needed based on their current source shares.
+First, every block, the amount of rewards to be distributed in that block are divided by the total source shares to get the rewards per share. This is added to a global total (named "global indexes"). This is repeated every block such that the global indexes represents the total rewards a user should be owed per source share if they had held a deposit from when the rewards were created.
 
-Users' rewards must be updated whenever their source shares change. This happens through hooks into other modules that run before deposits/borrows/supplies etc.
+Then, if a user has deposited (say into a CDP) at the very start of the chain (and never changed their deposit), their current reward balance can be calculated at any time $t$ as
+
+$$
+rewards_t = global\_indexes_t * source\_shares_t
```
Comment (drklee3) [inline x/incentive/spec/01_concepts.md:26]: formatting nit: Seems like underscores in these blocks are being converted to subscripts

File: x/incentive/spec/01_concepts.md [hunk 3]
```diff
@@ -4,23 +4,86 @@ order: 1
...
+
+Enumeration of input states:
+- `source_shares`, `global_indexes`, or `claim_indexes` should never be negative
+- `global_indexes` >= `claim_indexes`
+- `global_indexes` and `claim_indexes` can be positive or 0, where not existing in the store is counted as 0
+
+- `source_shares` are the value before the update (eg before a hard deposit)
+
+ | `global_indexes` | `claim_indexes` | `source_shares` | description                                                                                                                                                                                                                                                |
+ |------------------|-----------------|-----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
```
Comment (drklee3) [inline x/incentive/spec/01_concepts.md:60]: nice table on possible states 👍

File: x/incentive/spec/01_concepts.md [hunk 4]
```diff
@@ -4,23 +4,86 @@ order: 1
...
+
+In code, to avoid storing the entire history of a user's source shares and global index values, rewards are calculated on every source shares change and added to a reward balance
+
+$$
+rewards_t = reward\_balance_{t -10} + (global\_indexes_t - global\_indexes_{t-10}) * source\_shares_t
+$$
+
+Old values of $reward\_balance$ and $global\_indexes$ ares stored in a `Claim` object for each user.
+
+Listeners on external modules fire to update these values when source shares change. For example when a user deposits to hard, a method in incentive is called that calculates the rewards accrued since the last time their deposit changed. These must be called whenever source shares change, otherwise incorrect rewards will be distributed.
```
Comment (drklee3) [inline x/incentive/spec/01_concepts.md:48]: I think this could include updating the reward_indexes

Comment (rhuairahrighairidh) [reply x/incentive/spec/01_concepts.md]: I added some more details about how the claim object is updated

**Review by drklee3 — APPROVED** [review]

**Pull request merged**